vstudio/wnbdevents.xml

```diff
@@ -0,0 +1,58 @@
+﻿<?xml version='1.0' encoding='utf-8' standalone='yes'?>
+<instrumentationManifest
+    xmlns="http://schemas.microsoft.com/win/2004/08/events"
+    xmlns:win="http://manifests.microsoft.com/win/2004/08/windows/events"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://schemas.microsoft.com/win/2004/08/events eventman.xsd">
+    <instrumentation>
+        <events>
+            <provider
+                guid="{FFACC4E7-C115-4FE2-9D3C-80FAE73BAB91}"
+                messageFileName="%WINDIR%\System32\drivers\wnbd.sys"
+                name="WinBD"
+                resourceFileName="%WINDIR%\System32\drivers\wnbd.sys"
+                symbol="DriverControlGuid">
+                <channels>
+                    <importChannel
+                        chid="SYSTEM"
+                        name="System"/>
+                </channels>
+                <templates>
+                    <template tid="tid_load_template">
+                        <data name="LogLevel"
+                              inType="win:UInt32"
+                              outType="xs:unsignedInt"/>
+                        <data name="LineNumber"
+                              inType="win:UInt32"
+                              outType="xs:unsignedInt"/>
+                        <data name="FunctionName"
+                              inType="win:AnsiString"
+                              outType="xs:string"/>
+                        <data name="Message"
+                              inType="win:AnsiString"
+                              outType="xs:string"/>
+                    </template>
+                </templates>
+                <events>
+                    <event
+                        channel="SYSTEM"
+                        level="win:Informational"
+                        message="$(string.Event.EventMessage)"
+                        opcode="win:Start"
+                        symbol="Event"
+                        template="tid_load_template"
+                        value="1"/>
+                </events>
+            </provider>
+        </events>
+    </instrumentation>
+    <localization xmlns="http://schemas.microsoft.com/win/2004/08/events">
+        <resources culture="en-US">
+            <stringTable>
+                <string id="Event.EventMessage"
+                        value="WinBD Loaded"/>
```

Comment:
Can we use the actual message instead? Otherwise all log entries will contain the "WinBD Loaded" message, which is confusing.

Reply:
Whops forgot about the message will change it based on the event data.

We include the log level in the event data.

We could add an event based on the driver log level and check inside `WnbdLog` which event to send.
If we do the above another question would be which events would be logged when tracing is enabled and which are enabled by default.

Reply:
The log level is already configurable at the WNBD level. I'd still prefer to properly set the log level on the entry, not just the payload. An error shouldn't show up in the event log as an info message.

I think we can set Event.Level just like we set Event.EventMessage: https://docs.microsoft.com/en-us/windows-hardware/drivers/devtest/adding-event-tracing-to-kernel-mode-drivers.

Reply:
Unfortunately it can't, only the message is force to be overwritten. This is why I suggested we create an entry per type of log level.

Indeed the level of tracing is configurable at the driver level, but I was referring to the fact that no actual events show up in the event log at the moment, only on traces we start/stop.
To add events directly to event log we need to define them at different `opcodes`.

I was thinking to log only errors directly to the event log and the via traces. Do you agree?

Reply:
As discussed offline we are logging `WNBD_LVL_ERROR` and `WNBD_LVL_WARN` directly in event viewer and the rest will appear only when a trace is enabled on our ETW provider.